### PR TITLE
Preparation for being able to comment every public item

### PR DESCRIPTION
**Pull request by annando**
This is some work in progress. The goal is to be able to comment public items although the user hadn't subscribed to the author.

This will be a work in multiple steps. The first step is the needed change to the frontend. The next step is to change the item processing for being able to create the posts. The third step will be the change to the protocol implementation of Diaspora and OStatus to allow these posts. The last step then will be the change to the DFRN protocol to finally being able to do this for Friendica posts as well. (Which will be the hardest part)

Since this is a longer lasting project and since we are currently rewriting the code, I want to submit the first step, so that there won't be any future merge problems. The changed code shouldn't do any harm to existing functionality.

### CODE REVIEW

**Comment by annando** [comment]
I will add another commit to this PR. With this commit I finished step 2 and step 3 is working in a way that the comment arrives at the target.

There are other issues: You don't see these comment under your personal posts, you won't receive further posts to the thread on Diaspora posts and the display page isn't updated after doing the comment. Because of these issues the functionality is still deactivated.

**Comment by annando** [comment]
Current status:
* There is now an option to enable the commenting.
* You can comment public posts.
* The comment is distributed via Diaspora and OStatus.
* The thread is updated after the comment.

Known issues:
* You cannot reshare public posts.
* You cannot delete or edit your comment.
* You don't see these threads in the network page.
* You don't see these threads in the api.
* You don't get notifications for answers to such posts.
* On Diaspora posts we have to subscribe to such posts.

File: mod/display.php [hunk 1]
```diff
@@ -379,114 +342,81 @@ function display_content(App $a, $update = 0) {
...
+	}
+	$o .= conversation($a, $items, 'display', $update_uid);
 
-		// Limit the description to 160 characters
-		if (strlen($description) > 160) {
-			$description = substr($description, 0, 157) . '...';
-		}
+	// Preparing the meta header
+	require_once('include/bbcode.php');
+	require_once("include/html2plain.php");
```
Comment (MrPetovan) [inline mod/display.php]: Code standards: Please remove parentheses from language constructs and please choose a single quote style, preferably single.

**Comment by annando** [comment]
It doesn't make sense to make a query for "id" *and* "uid", since the id is unique.

I like to have separate calls. I don't like "OR" in SQL very much. Depending on how it is done, it can be very bad for the performance.

**Comment by MrPetovan** [comment]
I hear you, which brings me back to my initial question: did you always make to check permissions whenever you removed such conditions?

**Comment by annando** [comment]
The check for "allow_cid", "allow_gid" and so on is superfluous - it is all done with the "private" field. Normally nothing of my changes should have any impact on permissions and security. But this is another reason to currently keep some of the stuff behind a config setting.

**Comment by MrPetovan** [comment]
Part of #4117

**Pull request merged**